manifests/server/default_privileges.pp

```diff
@@ -62,15 +62,15 @@
   #
   # Port, order of precedence: $port parameter, $connect_settings[PGPORT], $postgresql::server::port
   #
-  if $port != undef {
+  if $port {
     $port_override = $port
-  } elsif $connect_settings != undef and 'PGPORT' in $connect_settings {
+  } elsif 'PGPORT' in $connect_settings {
     $port_override = undef
   } else {
     $port_override = $postgresql::server::port
   }
 
-  if $target_role != undef {
+  if $target_role {
```

Comment:
I think we need update [here](https://github.com/puppetlabs/puppetlabs-postgresql/blob/ece77516a4f9f4badd18e25c8b3ae974ef5be459/manifests/server/default_privileges.pp#L36) as well, as its being optional param we should also update its data type?